pallets/loans-ref/src/lib.rs

```diff
@@ -532,10 +533,11 @@ pub mod pallet {
 			};
 
 			let _count = Self::update_active_loan(pool_id, loan_id, |loan| {
-				let state = Self::find_write_off_state(pool_id, loan.maturity_date());
-				let limit = state.map(|s| s.status()).unwrap_or_else(|_| status.clone());
+				let rule = Self::find_write_off_rule(pool_id, loan)?;
+				let limit = rule.map(|r| r.status).unwrap_or_else(|| status.clone());
 
-				loan.write_off(&limit, &status)
+				loan.write_off(&limit, &status)?;
+				Ok(limit)
```

Comment:
This is a "bug" fix. The previous `admin_writen_off` event was not registering the correct status applied. No security issue here